native/common/jp_convert.cpp

```diff
@@ -14,10 +14,61 @@
    See NOTICE file for details.
  *****************************************************************************/
 #include "jpype.h"
+#include <math.h>
+#include <bitset>
 
 namespace
 {
 
+template <jvalue func(void *c) >
```

Comment:
This is a crude implementation of the "half" floating type.    There is no native type in Java to support this but it is possible that someone could pass a half floating in for promotion to either float or double.   This code is exercised in the test suite.